src/cache.rs

```diff
@@ -360,25 +356,21 @@ impl Cache {
                 let pages_dir = path.join(pages_dir_name);
 
                 if pages_dir.exists() {
-                    fs::remove_dir_all(&pages_dir).map_err(|e| {
-                        CacheError(format!(
-                            "Could not remove cache directory ({}): {}",
-                            pages_dir.display(),
-                            e
-                        ))
+                    fs::remove_dir_all(&pages_dir).with_context(|| {
+                        format!(
+                            "Could not remove the cache directory at {}",
+                            pages_dir.display()
+                        )
                     })?;
                 }
             }
         } else if path.exists() {
-            return Err(CacheError(format!(
+            return Err(anyhow!(
                 "Cache path ({}) is not a directory.",
                 path.display()
-            )));
+            ));
         } else {
-            return Err(CacheError(format!(
-                "Cache path ({}) does not exist.",
-                path.display()
-            )));
+            return Err(anyhow!("Cache path ({}) does not exist.", path.display()));
```

Comment:
This could probably be refactored to move the constraints up:

```rust
ensure!(
    path.exists(),
    "Cache path ({}) does not exist.",
    path.display()
);
ensure!(
    path.is_dir(),
    "Cache path ({}) is not a directory.",
    path.display()
);

// Delete old tldr-pages cache location as well if present
// TODO: To be removed in the future
for pages_dir_name in [TLDR_PAGES_DIR, TLDR_OLD_PAGES_DIR] {
// ...
```

What do you think?

Reply:
Clear improvement, yeah.